runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -43,50 +36,155 @@ void configureHeader();
 void insertNoOpCommand(unsigned int numPadBytes);
 }
 
-#define OSTREAM_OP(O_TYPE, TYPE) O_TYPE &operator<<(O_TYPE &os, const TYPE &s);
-
 namespace mlir::iree_compiler::AMDAIE {
-#define TO_STRING(TYPE) std::string to_string(const TYPE &t);
+struct TileLoc {
+  int col, row;
+  inline bool operator<(const TileLoc& rhs) const {
+    return std::tie(col, row) < std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator==(const TileLoc& rhs) const {
+    return std::tie(col, row) == std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator!=(const TileLoc& rhs) const { return !(*this == rhs); }
+
+  operator XAie_LocType() const { return XAie_TileLoc(col, row); }
+  TileLoc(XAie_LocType loc) : col(loc.Col), row(loc.Row) {}
+  TileLoc(int col, int row) : col(col), row(row) {}
+};
+
+enum class AMDAIEDevice : uint32_t {
+  xcvc1902 = 1,
+  xcve2302 = 2,
+  xcve2802 = 3,
+  npu = 4,
+  npu1 = 4,
+  npu1_1col = 5,
+  npu1_2col = 6,
+  npu1_3col = 7,
+  npu1_4col = 8,
+};
+
+enum class AMDAIETileType : uint8_t {
+  AIETILE = 0U,
+  SHIMNOC = 1U,
+  SHIMPL = 2U,
+  MEMTILE = 3U,
+  MAX = 4U
+};
+
+struct AMDAIEDeviceModel {
+  XAie_Config configPtr;
+  XAie_DevInst devInst;
+
+  explicit AMDAIEDeviceModel(uint8_t aieGen, uint64_t baseAddr,
+                             uint8_t colShift, uint8_t rowShift,
+                             uint8_t devNColumns, uint8_t devNRows,
+                             uint8_t memTileRowStart, uint8_t nMemTileRows,
+                             uint8_t nShimTileRows, int partitionNumCols,
+                             int partitionStartCol = 1, bool aieSim = false,
```

Comment:
I don't like default arguments, they get annoying when you want to change APIs, and buggy when 2 operands with defaults have the same type (bool and bool) -- I'd prefer if these were explicit if not too big a pain please

Reply:
Fair enough